src/traces/waterfall/index.js

```diff
@@ -0,0 +1,43 @@
+/**
+* Copyright 2012-2019, Plotly, Inc.
+* All rights reserved.
+*
+* This source code is licensed under the MIT license found in the
+* LICENSE file in the root directory of this source tree.
+*/
+
+
+'use strict';
+
+var Waterfall = {};
+
+Waterfall.attributes = require('./attributes');
+Waterfall.layoutAttributes = require('../bar/layout_attributes');
+Waterfall.supplyDefaults = require('./defaults');
+Waterfall.supplyLayoutDefaults = require('../bar/layout_defaults');
```

Comment:
Interesting. Should waterfall trace adhere to `barmode`, `bargroupgap` and `bargap`?

Having a group and an overlay `barmode` probably makes sense. I doubt we that `barmode: 'stack'` should be implemented.

Reply:
`barnorm` is probably useful too.

Now, maybe we should add a separate set of attribute just for waterfall traces e.g. `waterfallmode`, `waterfallgap` ...

I doubt that graphs with bar and waterfall on the same subplots are very common.

Reply:
Even group mode is extremely hard to read, but it would be much clearer if we made connecting lines - something like `connectors: {width, color, dash, mode: ('between'|'spanning')}`?

Dunno about the name and values for `mode` but I'm thinking `between` would be like this:
![between](https://upload.wikimedia.org/wikipedia/commons/9/96/Waterfall-chart.jpg)
and `spanning` like:
![spanning](https://upload.wikimedia.org/wikipedia/commons/3/3f/Waterfallchart_ex2.jpg)
My own preference would be for spanning solid black lines to be the default, because it makes clearer which end of the bar is the start and which is the end... not too important for it to be the default but I do think the connector feature should be there from the start.

There are also examples online of lines connecting the midpoints of the two bars, I'm not a fan but some people may want it:
![halfway connectors](https://cdn.vertex42.com/ExcelTemplates/Images/waterfall-chart-income-statement-breakdown.png)

FWIW `stack` does pop up here and there https://peltiertech.com/excel-waterfall-charts-bridge-charts/:
![stack](https://peltiertech.com/images/2016-09/StackedWaterfall.png)
But I don't see it as particularly important at least to start.